python/tests/kernel/test_run_kernel.py

```diff
@@ -1180,6 +1180,290 @@ def kernel_with_slots_dataclass() -> SlotsClass:
         e.value)
 
 
+def test_return_from_if_loop_with_true_condition():
```

Comment:
Can you add a test somewhere that dumps the kernel's Quake IR and uses FileCheck to verify it?